namer/marathon/src/main/scala/io/buoyant/namer/marathon/MarathonInitializer.scala

```diff
@@ -53,6 +64,24 @@ case class MarathonConfig(
 
   private[this] def getDst = dst.getOrElse(s"/$$/inet/$getHost/$getPort")
 
+  val secretKey = "DCOS_SERVICE_ACCOUNT_CREDENTIAL"
+  private[this] def getAuth = {
+    if (sys.env.contains(secretKey)) {
+      Api.readJson[MarathonSecret](Buf.Utf8(sys.env(secretKey))) match {
+        case Return(MarathonSecret(Some(loginEndpoint), Some(privateKey), Some("RS256"), Some(uid))) =>
+          Some(
+            Api.Auth(
+              new URL(loginEndpoint).getPath,
+              s"""{"uid":"$uid","token":"${Jwt.encode(s"""{"uid":"$uid"}""", privateKey, JwtAlgorithm.RS256)}"}"""
```

Comment:
string interpolation inside of string interpolation is crazy, yo
I'd break the token value into it's own val